src/charm.py

```diff
@@ -304,6 +306,27 @@ def _update_ingress_relation(self) -> None:
                 self._ingress_config, static=self._static_ingress_config
             )
 
+    def _update_tempo_api_relations(self) -> None:
+        """Update all applications related to us via the tempo-api relation."""
+        if not self.unit.is_leader():
+            return
+
+        # tempo-api should only send an external URL if it's set, otherwise it leaves that empty
+        internal_url = self._internal_url
+        external_url = self._external_url
+        if external_url == internal_url:
+            # external_url is not set and just defaulted back to internal_url.  Set it to None
+            external_url = None
+
+        tempo_api = TempoApiProvider(
+            relations=self.model.relations,
+            ingress_url=external_url,  # pyright: ignore
+            direct_url=internal_url,  # pyright: ignore
+            relation_name=TEMPO_API_RELATION_ENDPOINT_NAME,
```

Comment:
there's no harm in omitting the default if it matches. A unittest will catch any desync soon enough.